server/src/test/java/org/elasticsearch/index/mapper/AbstractScriptFieldTypeTestCase.java

```diff
@@ -186,8 +186,8 @@ protected static SearchExecutionContext mockContext() {
         return mockContext(true);
     }
 
-    protected static FieldDataContext mockFielddataContext() {
-        return new FieldDataContext("test", mockContext()::lookup);
+    protected static FieldDataContext mockFielddataContext(MappedFieldType.FielddataOperation fielddataOperation) {
+        return new FieldDataContext("test", mockContext()::lookup, mockContext()::sourcePath, fielddataOperation);
     }
```

Comment:
Given that these are script tests, I think we can just always use FielddataOperation.SCRIPT here?

Reply:
This is an interesting case for runtime fields. I don't think it's actually possible to hit source fallback for a runtime field at all, but they will likely be executed at a top level as a search operation, so I think for these tests it would make the most sense to switch it to SEARCH rather than SCRIPT.

Reply:
> I don't think it's actually possible to hit source fallback for a runtime field at all

I'm confused, runtime fields explicitly use scripts so I think they're quite likely to use this path? Or am I misunderstanding what's going on here?

Reply:
My understanding is when a runtime field is first executed as a script its part of a search query where the runtime field looks like a standard field. At that first execution the runtime field should always be found as the doc values portion of a field that it's emulating, right? So I don't think the operation actually matters for that "top-level" runtime field. However, for fields referenced from that runtime field then the operation will be SCRIPT when we do the lookup from LeafDocLookup.

Reply:
For the tests you're right, this should be SCRIPT. I'm thinking about this incorrectly.

Reply:
Fixed.